redis/src/connection.rs

```diff
@@ -167,7 +167,11 @@ impl IntoConnectionInfo for String {
 
 fn url_to_tcp_connection_info(url: url::Url) -> RedisResult<ConnectionInfo> {
     let host = match url.host() {
-        Some(host) => host.to_string(),
+        Some(host) => match host {
+            url::Host::Domain(path) => path.to_string(),
+            url::Host::Ipv4(v4) => v4.to_string(),
+            url::Host::Ipv6(v6) => v6.to_string(),
+        },
```

Comment:
Maybe add a comment here on why we can't use `host.to_string()` directly, to preserve your research?

Reply:
Sure, this issue is quite convoluted and should indeed be explained with enough comments.